src/main/java/io/cryostat/net/web/http/api/v2/TargetsPostHandler.java

```diff
@@ -148,7 +157,14 @@ public IntermediateResponse<ServiceRef> handle(RequestParameters params) throws
                 }
             }
             Map<AnnotationKey, String> cryostatAnnotations = new HashMap<>();
-            ServiceRef serviceRef = new ServiceRef(uri, alias);
+
+            String jvmId = null;
+            try {
+                jvmId = jvmIdHelper.getJvmId(uri.toString());
+            } catch (JvmIdGetException e) {
+                logger.warn("Couldn't put jvmId in serviceRef for {}", uri.toString());
```

Comment:
I think this should actually result in an error response, probably in the `4xx` range.